go/pkg/shim/shim.go

```diff
@@ -73,6 +74,7 @@ func SetVendorDB(shim string, db, dbx efi.SignatureDatabase) error {
 	if err != nil {
 		return err
 	}
+	defer os.Remove(fp.Name())
 
 	if err := VendorDBSectionWrite(fp, db, dbx); err != nil {
 		return err
```

Comment:
While you're here, could you fix the filehandle leak in the error path here?

ie, fp.Close() never gets called if we take this error path. (it gets called in the happy path one line below).